### DISCUSSION TITLE
Doc about trustStorePath and trustStorePass

### DISCUSSION THREAD

**Discussion by chanjarster** [apache/skywalking#8208] [General]
Currently there is a remark on `storage.elasticsearch.trustStorePath` and `trustStorePass`:

> Only works when username and password are enabled.

See below:

https://github.com/apache/skywalking/blob/1acba5d2467f6d905a6261e999f85464d5432562/docs/en/setup/backend/configuration-vocabulary.md#L91-L92

I think that's not accurate, should be:

* `trustStorePath` Trust JKS file path. Provide if `protocol` is `https`, the JKS file should contains Elasticsearch's CA certificate.
* `trustStorePass` Trust JKS file password. Provide if Trust JKS file needs a password.

**Comment by wu-sheng** [comment]
@hanahmily Your scope, I think.

**Comment by hanahmily** [comment]
There was a precondition that the username and password should be present before 8.8.0 https://github.com/apache/skywalking/blob/v8.7.0/oap-server/server-library/library-client/src/main/java/org/apache/skywalking/oap/server/library/client/elasticsearch/ElasticSearchClient.java#L167

The new restful-based elasticsearch client doesn't have such limitations.  I think we have a chance to update the document. 

From the current codes, the ES client intends to load them when `trustStorePath` exists, regardless of the protocol.

**Reply by chanjarster** [reply]
Maybe the code need to be changed. After all, `trustStorePath` only make sense if protocol is `https`, is it?

**Reply by wu-sheng** [reply]
HTTPS should be required.
